kernel/network/routing.h

```diff
@@ -0,0 +1,39 @@
+
+/******************************************************************************
+ *       ninjastorms - shuriken operating system                              *
+ *                                                                            *
+ *    Copyright (C) 2013 - 2016  Andreas Grapentin et al.                     *
+ *                                                                            *
+ *    This program is free software: you can redistribute it and/or modify    *
+ *    it under the terms of the GNU General Public License as published by    *
+ *    the Free Software Foundation, either version 3 of the License, or       *
+ *    (at your option) any later version.                                     *
+ *                                                                            *
+ *    This program is distributed in the hope that it will be useful,         *
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+ *    GNU General Public License for more details.                            *
+ *                                                                            *
+ *    You should have received a copy of the GNU General Public License       *
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.   *
+ ******************************************************************************/
+
+#pragma once
+
+#include "kernel/network/ethernet.h"
+
+#define ROUTING_DEBUG
+
+#define MAX_ARP_TABLE_ENTRIES 10
+
+struct __arp_table_entry {
+  uint32_t ip;
+  mac_address_t mac;
+};
```

Comment:
Do this struct need to be packed?

Reply:
Does not, as we store it in our memory and do not use it directly in packets